src/pages/take-quiz-page/take-quiz.tsx

```diff
@@ -1,7 +1,307 @@
+                            inputId={`answer${indexAns + 1}`}
+                            name="isCorrectAnswer"
+                            value={`answer${indexAns}`}
+                            onChange={() =>
+                              changeCheckedAnswer(question, indexAns)
+                            }
+                            checked={answer.isCorrectAnswer || false}
+                          />
+                        )}
+
+                        <label htmlFor={`answer${indexAns + 1}`}>
+                          {answer.answerName}
+                        </label>
+                      </div>
+                    </div>
+                  )}
+
+                  {question.questionType ===
+                    QUESTION_TYPES.SUBJECTIVE.value && (
+                    <InputTextarea
+                      style={{ width: "640px" }}
+                      className="take-quiz__question-response create-quiz__input"
+                      placeholder="Enter your answer"
+                      name={"inputText"}
+                      value={answer.answerName}
+                      onChange={(event) => {
+                        answer.answerName = event.target.value;
+                      }}
+                    />
+                  )}
+                </div>
+              ))}
+            </div>
+
+            <div className="take-quiz__divider" />
+
+            {/* FOOTER */}
+            <div className="take-quiz__footer flex flex-row justify-content-between">
+              <Button
+                disabled={index === 0}
+                onClick={() => {
+                  changeElement("previous");
+                }}
+                className="button__common-style button__take-quiz-previous"
+              >
+                <img src={previous_icon} alt="previous icon" />
+              </Button>
+
+              <div className="flex flex-row align-items-center gap-5">
+                {/* PAGE NUMBER */}
+                <div className="flex flex-row align-items-center gap-1">
+                  {`${index + 1} out of `}
+                  <div
+                    style={{ fontWeight: "700" }}
+                  >{` ${questions.length}`}</div>{" "}
+                </div>
+
+                <Button
+                  onClick={() => {
+                    if (index === questions.length - 1) {
+                      calculateFinalScore();
+                      navigate({
+                        pathname: "/congratulations",
+                        search: createSearchParams({
+                          quizId: `${quizId}`,
+                        }).toString(),
+                      });
+                    } else {
+                      changeElement("next");
+
+                      QuizStore.editUserQuestionAnswer(question.id, {
+                        ...question,
+                        answerList: questionAnswers,
+                      });
+                    }
+                  }}
```

Comment:
This function It's quite large, I suggest extracting it into a referenced function instead of using a lambda function.